dexmaker/src/main/java/com/android/dx/DexMaker.java

```diff
@@ -313,9 +314,11 @@ public void declare(FieldId<?, ?> fieldId, int flags, Object staticValue) {
      * Generates a dex file and returns its bytes.
      */
     public byte[] generate() {
-        DexOptions options = new DexOptions();
-        options.targetApiLevel = DexFormat.API_NO_EXTENDED_OPCODES;
-        DexFile outputDex = new DexFile(options);
+        if (null == outputDex) {
```

Comment:
I don't love all the new stateful-ness that this approach introduces. Is there a way we can add this functionality without it?

Reply:
Please see my explanation in a new comment

Reply:
It's done by new commit 733f075